### ISSUE TITLE
Restart on file changes

### ISSUE THREAD

**Issue by egoist** [egoist/poi#250]
```js
module.exports = {
  // watch poi.config.js and restart poi when it's changed
  restartOnFileChanges: true,
  // or custom files
  restartOnFileChanges: ['poi.config.js', 'package.json']
}
```

**Comment by egoist** [comment]
/cc @QingWei-Li 

![coming](https://media.giphy.com/media/PusssOtnPwT2o/giphy.gif)

**Comment by skiano** [comment]
Is the plan for this to work a bit like the vue-hacker-news demo code, where the server updates on changes to the template file: https://github.com/vuejs/vue-hackernews-2.0/blob/master/build/setup-dev-server.js#L34-L38 ? and does the comment above mean that  @QingWei-Li is already working on this?

**Comment by QingWei-Li** [comment]
@skiano
No, I'm not going to do it yet. 😅

**Comment by skiano** [comment]
@egoist
If I want to look into this, can i just npm link a local fork to a project and use it that way? or is there a simpler way to work?

**Comment by egoist** [comment]
@skiano yeah that's how I run it locally too

**Comment by skiano** [comment]
@egoist

Do you like the idea of using `chokidar` to watch the files array and telling webpack dev server to reboot when something changes? That's what I am considering

**Comment by egoist** [comment]
it's fine to use chokidar, but it actually needs to restart the whole process instead of the webpack-dev-server only, since the webpack config that is used to start webpack-dev-server might have changes too.

**Comment by skiano** [comment]
Ahh, makes sense. Do you have any idea where to begin with that, or is it totally unknown. I would like to help if possible, but I want to make sure I do not screw up all your great work or waste your time :)

**Comment by egoist** [comment]
https://github.com/egoist/poi/blob/master/packages/poi/bin/run.js only this file matters I think

**Comment by skiano** [comment]
Thanks, Ill take a look at that

**Comment by skiano** [comment]
@egoist this seems very tricky to me. Everything I try seems to hit a wall. Tearing down the webpack dev server cleanly from node seems like it does not work well. So I thought maybe I should be using the web pack dev middleware directly. But if I do that it means reconsidering way more than that one file, and it would also mean not getting to pass options to webpack dev server and instead just getting what comes with the middleware. So then I thought maybe I could just use something like nodemon to actually manage the process, but that felt icky. I even hit problems because the poi config is cached by node require, so even if I got wbepack to shut down on a file change, I still needed to invalidate that module in requires cache.

Basically, I am stumped, and I wanted to let you know

**Comment by egoist** [comment]
@skiano this is what I did previously: https://github.com/egoist/poi/blob/f00d79eb60c00d509e15d02ec4ada6d5ec9ed770/bin/vbuild-start.js#L102-L133 wrap the whole thing in a `start` function, watch the config and re-`start` on file changes.

**Comment by skiano** [comment]
@egoist Interesting! So why did you not just use that?

**Comment by egoist** [comment]
@skiano haven't got time for this 😅

**Comment by skiano** [comment]
Makes sense! I just saw you are also doing a bunch for vue-cli 3!!!! when do you find the time :) Sorry to bug you.

**Comment by egoist** [comment]
Added https://github.com/egoist/poi/releases/tag/poi%409.5.1
